protocols/kad/src/protocol.rs

```diff
@@ -294,6 +308,22 @@ pub enum KadResponseMsg {
         /// Known providers for this key.
         provider_peers: Vec<KadPeer>,
     },
+
+    /// Response to a `GetValue`.
+    GetValue {
+        /// Result that might have been found
+        result: Option<(Multihash, Vec<u8>)>,
+        /// Nodes closest to the key
+        closer_peers: Vec<KadPeer>,
+    },
+
+    /// Response to a `PutValue`.
+    PutValue {
+        /// The key of the record.
+        key: Multihash,
+        /// The value of the record.
+        value: Vec<u8>,
+    },
```

Comment:
See the other comment about whether there is actually supposed to be a response to a `PUT_VALUE` and if so whether it carries any data.